pkg/client/util/http_backoff.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+// This is a custom Backoff that enforces the Max wait duration.
+// If the sleep is greater we refuse to sleep at all
+func HTTPBackOff(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
+	sleep := retryablehttp.DefaultBackoff(min, max, attemptNum, resp)
+	if sleep.Abs() <= max {
+		return sleep.Abs()
```

Comment:
Minor code legibility - this is backwards to how I'd normally read code. I'd normally expect the conditional to handle the exception case, here we're returning what I'd expect would be the normal mode of operation.